tasks/Module 3/task_2.py

```diff
@@ -0,0 +1,57 @@
+"""Module2/task2: custom, python split function."""
+import re
+import pytest
+
+# Vars
+EMPTY_STRING_ERROR = 'Please provide a valid separator. ' \
+                     'It cannot be an empty string.'
+
+
+# Test data
+TEST_DATA_SET_1 = [
```

Comment:
Nice usage of parametrize for these scenarios, but I would add some more test cases here :) Especially I am missing a case then separator is not provided. You can add for example:
```python
('string\twith  double  spaces and\nnew\nlines', None, ['string', 'with', 'double', 'spaces', 'and', 'new', 'lines'])
```